ement-room.el

```diff
@@ -2817,151 +2809,113 @@ MSG is the message STRING was contained in."
                                     'action #'browse-url-at-mouse
                                     'follow-link t)
                if (and ement-room-url-previews
+                       (not (map-elt (map-elt (ement-event-content msg) 'm.relates_to)
+                                     'm.in_reply_to))
                        ;; spec recommends not getting previews for encrypted rooms
                        (not (with-current-buffer buffer
                               (cl-find-if (apply-partially #'equal "m.room.encryption")
                                           (ement-room-invite-state ement-room)
                                           :key #'ement-event-type))))
-               do (ement-room--get-preview buffer
-                                           msg
-                                           (buffer-substring-no-properties (match-beginning 0)
-                                                                           (match-end 0))
-                                           c)
+               do (let ((string (buffer-substring-no-properties (match-beginning 0) (match-end 0))))
```

Comment:
I feel like doing this in `--linkify-urls` is an abuse of the function.  :)  This function is intended to do one thing: return the string argument with any URLs in it having been linkified.

Reply:
Would you suggest creating a separate filter function instead (though that would mean parsing the string for urls twice)? Maybe renaming linkify-urls as a more general url filter handler is a better option? Because it currently seems to be used only once in one place in the existing code, so...

Reply:
This function does one thing.  If we need a function to do something else, it should be separate.